torch/cuda/streams.py

```diff
@@ -112,6 +112,29 @@ def __repr__(self):
                 .format(self.device, self.cuda_stream))
 
 
+class ExternalStream(Stream):
+    r"""Wrapper around a externally allocated CUDA stream.
+
+    A CUDA stream is a linear sequence of execution that belongs to a specific
+    device, independent from other streams.  See :ref:`cuda-semantics` for
+    details.
+
+    This class is used to wrap streams allocated in other libraries in order
+    to facilitate data exchange and multi-library interactions. Be aware
+    that the stream life-cycle is managed by the owner.
+
+    Args:
+        stream_ptr(int): Integer representation of the `cudaStream_t` value.
+            allocated externally.
+        device(torch.device or int, optional): the device where the stream
```

Comment:
we should mention that if device was specified incorrectly, subsequent launches using this stream may fail. Unfortunately, there's no way to verify if device is correct or not.

Reply:
noted!